services/port_manager/src/main/java/com/futurewei/alcor/portmanager/service/PortServiceImpl.java

```diff
@@ -296,4 +295,9 @@ public RouterUpdateInfo updateL3Neighbors(String projectId, RouterUpdateInfo rou
 
         return routerUpdateInfo;
     }
+
+    @Override
+    public SubnetPortIds getSubnetPorts(String projectId, String subnetId) throws Exception {
+        return portRepository.getSubnetPortIds(subnetId);
+    }
```

Comment:
@chenpiaoping This API just need to return # of ports in the subnet, you don't need to return their port ids. Subnet Manager will check this return value when the subnet is going to be deleted.

Reply:
fixed, please review again.